.github/workflows/test_and_notify_faliure.yaml

```diff
@@ -0,0 +1,88 @@
+
+on:
+  inputs:
+    integration-test-extra-arguments:
+      description: Additional arguments to pass to the integration tests
+      type: string
+    integration-test-pre-run-script:
+      description: Path to the bash script to be run before the integration tests
+      type: string
+    integration-test-provider:
+      description: Actions operator provider for the integration tests as per https://github.com/charmed-kubernetes/actions-operator#usage
+      type: string
+      default: microk8s
+    integration-test-series:
+      description: List of series to run the integration tests in JSON format, i.e. '["jammy", "focal"]'. Each element will be passed to tox as --series argument
+      type: string
+      default: '[""]'
+    email-server-address:
+      description: Address of email server used for sending email
+      type: string
+      default: smtp.gmail.com
+    email-server-port:
+      description: Port of email server used for sending email
+      type: int
+      default: 465
+    email-receiver:
+      description: Email address for receiving the failure notification.
+      type: string
+
+jobs:
+  lib-check:
+    # This job is the precondition of all jobs that need the Charmhub token.
+    # By disabling this job on forked repositories, we can achieve
+    # disabling the publishing to Charmhub action on forked repositories
+    # while still running tests on push events
+    if: github.repository_owner == 'canonical'
+    name: Check libraries
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Check libs
+        uses: canonical/charming-actions/check-libraries@2.1.1
+        with:
+          credentials: ${{ secrets.CHARMHUB_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+  tests:
+    name: Run tests
+    uses: ./.github/workflows/test.yaml
+  integration-tests:
+    name: Run integration tests
+    uses: ./.github/workflows/integration_test.yaml
+    with:
+      extra-arguments: ${{ inputs.integration-test-extra-arguments }}
+      pre-run-script: ${{ inputs.integration-test-pre-run-script }}
+      provider: ${{ inputs.integration-test-provider }}
+      series: ${{ inputs.integration-test-series }}
+    secrets: inherit
+  notify-on-failure:
+    name: Notify on failure
+    needs: [integration-test]
+    if: ${{ inputs.notify && failure() }}
+    runs-on: ubuntu-latest
+    env:
+      USERNAME: ${{ secrets.EMAIL_USERNAME }}
+      PASSWORD: ${{ secrets.EMAIL_PASSWORD }}
+    steps:
+      - name: Check credential exists
+        if: ${{ env.USERNAME == null || env.PASSWORD == null }}
+        run: |
+          echo "Missing email username and password as input for this workflow."
+          exit 1
+      - name: Notify Failure
+        uses: dawidd6/action-send-mail@v3
+        with:
+          server_address: ${{ inputs.email-server-address }}
+          server_port: ${{ inputs.email-server-port }}
+          username: ${{ env.USERNAME }}
```

Comment:
Is it possible to replace this with `${{ secrets.EMAIL_USERNAME }}` rather than having to pass it through the environment? It would simplify this a bit and also only provide the secrets specifically where they are needed

Reply:
I have updated this.